contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -8677,12 +8688,60 @@ handleGeospatialFunctionsInFunctionCall(TSqlParser::Function_callContext *ctx)
 
 		/* This if-elseIf clause rewrites the query in case of geospatial function calls */
 		if (ctx->spatial_proc_name_server_database_schema()->geospatial_func_arg() && ctx->function_arg_list())
-			rewrite_function_call_geospatial_func_ref_args(ctx);
+		{
+			size_t col_stop_index = ctx->spatial_proc_name_server_database_schema()->column->stop->getStopIndex();
+			size_t func_start_index = ctx->spatial_proc_name_server_database_schema()->geospatial_func_arg()->start->getStartIndex();
+			size_t arg_list_start_index = ctx->function_arg_list()->start->getStartIndex();
+			size_t arg_list_stop_index = ctx->function_arg_list()->stop->getStopIndex();
```

Comment:
why are we doing this inside rewrite_function_call_dot_func_ref_args?

Reply:
moved this index calculation inside `rewrite_function_call_dot_func_ref_args` function